source/common/upstream/health_discovery_service.cc

```diff
@@ -54,12 +59,51 @@ void HdsDelegate::onReceiveInitialMetadata(Http::HeaderMapPtr&& metadata) {
   UNREFERENCED_PARAMETER(metadata);
 }
 
+void HdsDelegate::processMessage(
+    std::unique_ptr<envoy::service::discovery::v2::HealthCheckSpecifier>&& message) {
+  // Create HdsCluster config
+  ENVOY_LOG(debug, "Creating HdsCluster config");
+  envoy::api::v2::core::BindConfig bind_config;
+  // envoy::api::v2::Cluster& cluster_config_ = cluster_config;
+  cluster_config_.set_name("anna");
+  cluster_config_.mutable_connect_timeout()->set_seconds(2);
+  cluster_config_.mutable_http2_protocol_options();
+  cluster_config_.mutable_per_connection_buffer_limit_bytes()->set_value(12345);
+
+  // Add endpoint to cluster
+  cluster_config_.add_hosts()->MergeFrom(*message->mutable_health_check(0)
+                                              ->mutable_endpoints(0)
+                                              ->mutable_endpoints(0)
+                                              ->mutable_address());
+
+  // Add health checker to cluster
+  cluster_config_.add_health_checks()->MergeFrom(
+      *message->mutable_health_check(0)->mutable_health_checks(0));
+
+  ENVOY_LOG(debug, "New HdsCluster config {} ", cluster_config_.DebugString());
+
+  // Creating HdsCluster
+  cluster_.reset(new HdsCluster(runtime_, cluster_config_, bind_config, store_stats,
+                                ssl_context_manager_, secret_manager_, false));
+
+  // Creating HealthChecker
+  health_checker_ptr = Upstream::HealthCheckerFactory::create(
+      *cluster_config_.mutable_health_checks(0), *cluster_, runtime_, random_, dispatcher_);
+}
+
 void HdsDelegate::onReceiveMessage(
     std::unique_ptr<envoy::service::discovery::v2::HealthCheckSpecifier>&& message) {
-  ENVOY_LOG(debug, "New health check response ", message->DebugString());
+
   stats_.requests_.inc();
-  stream_->sendMessage(health_check_request_, false);
-  stats_.responses_.inc();
+  ENVOY_LOG(debug, "New health check response message {} ", message->DebugString());
+
+  // Process the HealthCheckSpecifier message
+  processMessage(std::move(message));
+
+  // Initializing Cluster & HealthChecker
+  cluster_->setHealthChecker(health_checker_ptr);
+  bool initialized = false;
+  cluster_->initialize([&initialized] { initialized = true; });
```

Comment:
You probably don't need the initialized callback here (it can be a nop).

Reply:
Done